dashboard/app/controllers/api_controller.rb

```diff
@@ -184,8 +184,16 @@ def section_progress
 
     script_levels = script.script_levels.where(bonus: nil)
 
+    # Clients are seeing requests time out for large sections as we attempt to
+    # send back all of this data. Allow them to instead request paginated data
+    if params[:page] && params[:per]
+      paged_students = section.students.page(params[:page]).per(params[:per])
```

Comment:
For posterity: We're using the [`kaminari` gem](https://github.com/kaminari/kaminari) for pagination here.